examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -203,30 +203,52 @@ def highlight_min(x):
 # We can also evaluate the ability of the two extreme quantile estimators at
 # producing a well-calibrated conditional 90%-confidence interval.
 #
-# To do this we can compute the fraction of observations that fall between the
-# predictions:
+# To do this we can define two functions:
+#
+# - one to compute the fraction of observations that fall between the
+#   predictions;
+# - one to wrap those metrics in a cross-validation loop and assess their
+#   variability under data resampling.
+
+from sklearn.model_selection import KFold
+
+
 def coverage_fraction(y, y_low, y_high):
     return np.mean(np.logical_and(y >= y_low, y <= y_high))
 
 
-coverage_fraction(
-    y_train,
-    all_models["q 0.05"].predict(X_train),
-    all_models["q 0.95"].predict(X_train),
+def cross_val_coverage_fraction(models_dict, X, y, cv):
+    coverage_results = []
+    for _, test_index in cv.split(y):
+        y_low = models_dict["q 0.05"].predict(X[test_index])
+        y_high = models_dict["q 0.95"].predict(X[test_index])
+        y_true = y[test_index]
+
+        coverage = coverage_fraction(y_true, y_low, y_high)
+        coverage_results.append(coverage)
+
+    mean_coverage = np.mean(coverage_results)
+    std_coverage = np.std(coverage_results)
+
+    return mean_coverage, std_coverage
```

Comment:
This function does not really do cross-validation, i.e., it does not retrain the models. It is just a simple (or poor) bootstrap estimator of mean and variance. For that we could also use `scipy.stats.bootstrap` or use the standard formula for the variance - no bootstrapping / resamplling.